core/src/interfaces.rs

```diff
@@ -251,6 +305,9 @@ pub struct ProofRequestOpt {
     pub proof_type: Option<String>,
     /// Blob proof type.
     pub blob_proof_type: Option<String>,
+    #[arg(long, require_equals = true)]
+    /// The guest image id for RISC0/SP1 provers.
+    pub image_id: Option<String>,
```

Comment:
I just feel it's not good to expose image_id in user's API, otherwise we need to deal with user's random inputs.

maybe put it into prover_args that is a anonymous key-value hashmap is better? And in follow logic, we just insert the correct image_id (as only we know the correct one).

Reply:
I will keep the type definition, but `assert!(image_id.is_none())` when HTTP API receives requests.

Reply:
[4031f19](https://github.com/taikoxyz/raiko/pull/442/commits/4031f19531a28d450d38787e24f9f50337f52e0d) Done

Reply:
1 more thing maybe not directly related to this one. 
In future's ZK_ANY mode, we probably still need to put image_id to zk provider's specific args if we need to support customer image_id.